### PR TITLE
refactor error messages & remove logging

### PR DESCRIPTION
**Pull request by Nesze**
1. Logging to the default log.Logger is not cool, as clients may override the flags and patterns.
2. Error logs should not be spread across different log messages. 
3. The client should be responsible for logging the returned error messages. 

Adding support for logging file/method/line number could follow, this way we could even have stacktrace-like error logging.

Thoughts?

### CODE REVIEW

File: consumer/consumer.go [hunk 1]
```diff
@@ -82,19 +92,19 @@ func (c *DefaultIterator) consume() ([]Message, error) {
 	if c.consumer == nil {
 		cInst, err := q.createConsumerInstance()
 		if err != nil {
-			log.Printf("ERROR - creating consumer instance: %s", err.Error())
+			err = fmt.Errorf("Error creating consumer instance: [%v]", err)
```
Comment (martingartonft) [inline consumer/consumer.go]: why not just return the error instead of wrapping it?

Comment (martingartonft) [inline consumer/consumer.go]: same question for the other instances.

Comment (Nesze) [inline consumer/consumer.go]: For better traceability. Not that it is hard to track, though...

**Comment by Zapadlo** [comment]
+1

**Comment by Nesze** [comment]
Orphan PR.

**Pull request closed without merging**